src/utils/checkInDup.ts

```diff
@@ -0,0 +1,26 @@
+import { google } from "googleapis";
+import { CHECKIN_SHEET_ID } from "../controllers/checkins";
+import { CheckIn } from "../models/checkin";
+import { serialize_rows, sheet_auth } from "../utils/sheets";
+
+export const check_dup_student = async (eventId, studentId) => {
+    const sheet = google.sheets("v4");
+  
+    const read_result = await sheet.spreadsheets.values.get({
+      spreadsheetId: process.env.SHEET_ID,
+      auth: sheet_auth(),
+      range: `${CHECKIN_SHEET_ID}!A1:D`
+    });
+  
+    const rows = read_result.data.values as string[][];
+    const ser = serialize_rows(rows) as Array<CheckIn>;
+
+    for (let i = 0; i < ser.length; i++) {
+        const event = ser.at(i)?.event
```

Comment:
Javascript has some nice functions for this, and we should prefer to use them instead.

`return ser.findIndex(check_in => check_in.event === eventId && check_in.student_id === studentId) !== -1`